src/renderer/components/+cluster-settings/cluster-settings.tsx

```diff
@@ -49,20 +61,94 @@ export class ClusterSettings extends React.Component<Props> {
     }
   };
 
-  render() {
-    const cluster = this.cluster;
+  onTabChange = (tabId: string) => {
+    this.activeTab = tabId;
+  };
 
-    if (!cluster) return null;
-    const header = (
+  renderNavigation() {
+    return (
       <>
-        <h2>{cluster.preferences.clusterName}</h2>
+        <h2>{this.cluster.name}</h2>
+        <Tabs className="flex column" scrollable={false} onChange={this.onTabChange} value={this.activeTab}>
+          <div className="header">Cluster Settings</div>
+          { clusterPages.map((page) => {
+            return <Tab key={page} value={page} label={page} data-testid={`${page}-tab`} />;
+          })}
+        </Tabs>
       </>
     );
+  }
+
+  render() {
+    const cluster = this.cluster;
+
+    if (!cluster) return null;
 
     return (
-      <PageLayout className="ClusterSettings" header={header} showOnTop={true}>
-        <Status cluster={cluster}></Status>
-        <General cluster={cluster}></General>
+      <PageLayout
+        className="ClusterSettings"
+        navigation={this.renderNavigation()}
+        showOnTop={true}
+        contentGaps={false}
+      >
+        {this.activeTab == "General" && (
```

Comment:
This doesn't seem very extendable. Would it be better to render all of them and then switch one one is "active" using `css-grid` or something?

Reply:
I was mainly following how "Preferences" is implemented. I will open more issues related to this page once we have the layout in place (this PR merged) ... extensions should be able to hook into this page with new tabs.

Reply:
So we will be moving to that sort of design later?

Reply:
I was talking about dynamic "tabs" on PageLayout.. currently PageLayout is quite static but follow-up PR should add extension api that allows to add items into "catalog entity settings page". If that requires changes to PageLayout then we have to move... but I'm not sure yet.

Reply:
I meant more how we are rendering cluster settings, since I think that it is possible for extensions to add them but we are no longer rendering them.

Reply:
Ok, that will change. In the past extensions could only add "cluster features" but now they should be able to add almost anything into this page. ("cluster features" were removed already in #2418 ).

Reply:
Refactored to "catalog entity settings" -> pluggable.